src/conn/conn_tiered.c

```diff
@@ -96,7 +101,23 @@ __flush_tier_once(WT_SESSION_IMPL *session, uint32_t flags)
      * - Do the work to create said objects.
      * - Move the objects.
      */
-    S2C(session)->flush_state = 0;
+    conn->flush_state = 0;
+
+    /*
+     * Update time value for most recent flush_tier, not letting it move backwards and making sure
+     * it is after the most recent checkpoint. We hold the checkpoint lock so we know no other
+     * thread can be doing a checkpoint at this time but our time can move backward with respect to
+     * the time set by a different thread that did a checkpoint.
+     */
+    WT_ASSERT(session, FLD_ISSET(session->lock_flags, WT_SESSION_LOCKED_CHECKPOINT));
+    for (;;) {
+        __wt_seconds(session, &flush_time);
+        if (flush_time > conn->ckpt_most_recent)
+            break;
+        /* Sleep a smaller amount to hope we can start work sooner. */
+        __wt_sleep(0, 250000);
+    }
+    conn->flush_most_recent = flush_time;
```

Comment:
It would be nice if we could just set `conn->flush_most_recent` to `conn->ckpt_most_recent`.  It would save the fuss with making sure `wt_seconds()` returns a value that isn't too low.  It could also be done atomically which would (I think) eliminate the need for the checkpoint lock here.

But see my other comment about why the world doesn't seem that simple.